### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by paveltomin**
should go after https://github.com/GEOS-DEV/GEOS/pull/3100

- trying to remove some duplications in well solvers
- remove some duplication in reservoir and wells solvers

in both cases - just move what possible to the base class

### CODE REVIEW

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp
Comment (tjb-ltk) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp:1]: why return 0

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp:1]: This is just for consistency with the multiphase version, to be able to use them uniformly

File: src/coreComponents/physicsSolvers/fluidFlow/wells/SinglePhaseWell.cpp
Comment (tjb-ltk) [inline src/coreComponents/physicsSolvers/fluidFlow/wells/SinglePhaseWell.cpp:1]: at some point we could rename mixture connectionrate (compositional version) to connectionrate and move to well solver base...  they are the same size...

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp
Comment (tjb-ltk) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp:1]: fair enough.. at some point this needs to be rationalized.....the client could get this value other ways or the return a struct  to include other quants

File: src/coreComponents/physicsSolvers/multiphysics/CoupledReservoirAndWellsBase.hpp
Comment (tjb-ltk) [inline src/coreComponents/physicsSolvers/multiphysics/CoupledReservoirAndWellsBase.hpp:1]: nice cleanup and reorg... clarifies the which solver is being called  and

**Review by tjb-ltk — APPROVED** [review]

**Pull request merged**